numpy/ma/core.py

```diff
@@ -4266,21 +4290,26 @@ def count(self, axis=None):
 
         """
         m = self._mask
-        s = self.shape
         if m is nomask:
+            # see _count_reduce_items in _methods.py
             if axis is None:
-                return self.size
-            else:
-                n = s[axis]
-                t = list(s)
-                del t[axis]
-                return np.full(t, n, dtype=np.intp)
-        n1 = np.size(m, axis)
-        n2 = np.sum(m, axis=axis, dtype=np.intp)
-        if axis is None:
-            return (n1 - n2)
-        else:
-            return narray(n1 - n2)
+                axis = tuple(range(self.ndim))
+            if not isinstance(axis, tuple):
+                axis = (axis,)
+            items = 1
+            for ax in axis:
+                items *= self.shape[ax]
+            if kwargs.get('keepdims', False):
+                out_dims = list(self.shape)
+                for a in axis:
+                    out_dims[a] = 1
+                return items*ones(out_dims)
+            return items
+        if self is masked:
+            return 0
```

Comment:
This is incorrect since you ignore the keepdims, etc. (not a huge deal, but...) how about making it a factor or so? Use the nomask branch, but multiply the result by 0 or 1 depending on which branch you have.